lib/config/config-file.js

```diff
@@ -329,11 +329,7 @@ function applyExtends(config, filePath) {
             debug("Loading " + parentPath);
             return ConfigOps.merge(load(parentPath), previousValue);
         } catch (e) {
-            // If the file referenced by `extends` failed to load, add the path to the
-            // configuration file that referenced it to the error message so the user is
-            // able to see where it was referenced from, then re-throw
-            e.message += "\nReferenced from: " + filePath;
-            throw e;
+            return {}; // suppress error for unsupported extensions
```

Comment:
> That's my thinking.

Care to expand on that? :)

> An alternative approach

If it's functionally equivalent to white list good extensions (or even blacklist bad ones), and that list is reasonable to manage now and in the future, I'd much prefer that to silently swallowing all errors when loading config files.

My general thought is that if "we only support these extensions" or "these extensions are not permitted" is our actual intent here (as your commit message insinuates), then it'd be great if the code conveyed that better.